src/frontend/Sider/Behandling/VedtakOgBeregning/Barnetilsyn/utils.ts

```diff
@@ -30,3 +30,15 @@ export const lagVedtakRequest = (
         resultatType: BehandlingResultat.INNVILGET,
     };
 };
+
+export const barnSomOppfyllerAlleVilkår = (vilkår: Vilkårsvurdering) => {
+    const barnSomOppfyllerVilkår = vilkår.vilkårsett
+        .filter(
+            (vurdering) =>
+                vurdering.resultat === Vilkårsresultat.OPPFYLT ||
+                vurdering.resultat === Vilkårsresultat.AUTOMATISK_OPPFYLT
+        )
+        .map((vurdering) => vurdering.barnId);
+
+    return vilkår.grunnlag.barn.filter((barn) => barnSomOppfyllerVilkår.includes(barn.barnId));
```

Comment:
Skjønner ikke helt hvorfor denne filtreringen er nødvendig 🤔  Kan det finnes barni vilkår som ikke ligger i grunnlag?

Reply:
Kan finnes barn i grunnlaget som ikke har oppfylte vilkår. I grunnlaget har man alle barna som er med i behandlingen, men vi vil kun vise barna man faktisk kan få dekt utgiftene til

Reply:
Og så er det to ulike lister: I grunnlaget er den faktiske informasjonen om barnet, mens i vilkårene er det bare identen. Så det som skal skje her er: 

1. Finn ident på barn som har oppfylte vilkår - `barnSomOppfyllerVilkår` = liste med identer på ban som har resultat = oppfylt på pass_barn. 
2. Finn mer informasjon om barna som har oppfylte vilkår - gjøres ved å ta alle barna i grunnlaget (fordi der ligger navn og alder osv) og behold kun de som har oppfylte vilkår